### PR TITLE
Add tweet news in terminal toolbar

### PR DESCRIPTION
**Pull request by DidierRLopes**
Let me know if there's anything else you want me to do!

Screenshot 2022-12-08 at 22 40 39

### CODE REVIEW

**Comment by jmaslek** [comment]
Same thing is happening to me that was happening on the mstarpy branch.

```
Package operations: 2 installs, 80 updates, 2 removals

  • Removing setuptools (65.4.1)
  • Removing wheel (0.38.4)
```

@Chavithra @piiq Do we need to add setuptools to poetry?

**Review by piiq — CHANGES_REQUESTED** [review]
Please:
1. Make sure that there is exception handling in place for situations when data is queried for non-popular instruments
2. Make sure that timeouts for people who don't have twitter keys or are in networks where twitter is blocked are 1 second and less
3. Add unit tests for every function that you have added

**Comment by piiq** [comment]
@jmaslek we have setuptools in poetry build system. We need an older version to support building dependencies that rely on numpy
image

**Comment by jmaslek** [comment]
> @jmaslek we have setuptools in poetry build system. We need an older version to support building dependencies that rely on numpy image

So then why does poetry remove it -- but only on certain branches?

**Comment by DidierRLopes** [comment]
> Please:
> 
> 1. Make sure that there is exception handling in place for situations when data is queried for non-popular instruments
> 2. Make sure that timeouts for people who don't have twitter keys or are in networks where twitter is blocked are 1 second and less
> 3. Add unit tests for every function that you have added

> 1. Make sure that there is exception handling in place for situations when data is queried for non-popular instruments

What do you mean by "when data is queried for non-popular instruments"? This displays a tweet, unrelated with any instruments?

> 2. Make sure that timeouts for people who don't have twitter keys or are in networks where twitter is blocked are 1 second and less

Screenshot 2022-12-13 at 14 21 58

I already had a check to disable it in case the user didn't have correct API key. The timeout I left the default one as 60s, but can do 5s instead. 👍

> 3. Add unit tests for every function that you have added

Can you give me an example of what you expect the unit test for `update_news_from_tweet_to_be_displayed` to look like? Need to mockup the tweets of several users?

**Comment by piiq** [comment]
Re-solved dependencies here break the dep tree and poetry can't install on a fresh env. Hold on the merge please until this is resolved

File: openbb_terminal/helper_funcs.py [hunk 1]
```diff
@@ -65,11 +81,16 @@
...
 MENU_RESET = 2
 
+NEWS_TWEET = ""
+LAST_TWEET_NEWS_UPDATE_CHECK_TIME = datetime.now() - timedelta(hours=1)
+
 # Command location path to be shown in the figures depending on watermark flag
 command_location = ""
 
+NEWS_TWEET = ""
+LAST_TWEET_NEWS_UPDATE_CHECK_TIME = datetime.now() - timedelta(hours=1)
```
Comment (jmaslek) [inline openbb_terminal/helper_funcs.py:91]: dupe

File: openbb_terminal/helper_funcs.py [hunk 2]
```diff
@@ -1953,6 +1972,100 @@ def str_date_to_timestamp(date: str) -> int:
...
+    -------
+    str
+        The news from tweet to be displayed
+    """
+    global LAST_TWEET_NEWS_UPDATE_CHECK_TIME
+
+    # Check whether it has passed a certain amount of time since the last news update
+    if (
+        datetime.now(pytz.utc) - LAST_TWEET_NEWS_UPDATE_CHECK_TIME
+    ).seconds > obbff.TOOLBAR_TWEET_NEWS_SECONDS_BETWEEN_UPDATES:
```
Comment (jmaslek) [inline openbb_terminal/helper_funcs.py:1988]: You probably want to use total_seconds here instead.  Seconds will only give you a number from 0-60 based on the timestamps.

File: openbb_terminal/miscellaneous/scripts/economy/test_economy.openbb [hunk 3]
```diff
@@ -0,0 +1,86 @@
+economy
```
[the remark this answers is not in the document]
Comment (Chavithra) [reply openbb_terminal/miscellaneous/scripts/economy/test_economy.openbb]: It was moved then changed.
Deleted the file.

**Comment by jmaslek** [comment]
https://techcrunch.com/2023/02/01/twitter-to-end-free-access-to-its-api/?guccounter=1&guce_referrer=aHR0cHM6Ly93d3cuYmluZy5jb20v&guce_referrer_sig=AQAAAIfbpM1o0f_hqTCF2yXUIeoiC3ccQaWgV2iFFqWJOffvqMdRQbVo8852Gxr578cE9oC3pd48mfMNkuFRFx2IzS0M4vF3S7HMCp1OC0bG7fnudoOMop-nyUuy2MZmAWb8NIGn6DsRGqViA6dLfEJr1Dqv0K9HeUVkH7tiekz67qCQ

**Review by Chavithra — APPROVED** [review]
Looks ok to me.

There is a API_TWITTER_BEARER_TOKEN loading issue : this feature requires the credentials to be loaded sooner.

But I think we should try to resolve that on the credentials refactoring that is planned.

For now it works if we put the credentials on `config_terminal.py`.

**Pull request merged**